build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -85,11 +84,12 @@ class BuildImpl {
         _verbose = options.verbose,
         _environment = buildDefinition.environment,
         _trackPerformance = options.trackPerformance,
-        _buildDirs = options.buildDirs,
         _logPerformanceDir = options.logPerformanceDir;
 
-  Future<BuildResult> run(Map<AssetId, ChangeType> updates) =>
-      _SingleBuild(this).run(updates)..whenComplete(_resolvers.reset);
+  Future<BuildResult> run(
+          Map<AssetId, ChangeType> updates, List<String> buildDirs) =>
```

Comment:
Why not make `buildDirs` an optional (named) argument instead (with [] as the default)? Then i don't think we have to deal with a breaking change (and its a sane default anyways).

Reply:
I think it would still be a breaking behavior change

Reply:
This is still a breaking change regardless as we expose `BuildOptions`. I do like making the buildDirs optional though.